### PR TITLE
feat(eslint): add Testing Library plugin for ESLint

### PR DESCRIPTION
**Pull request by larimaza**
## Purpose

Add [Testing Library plugin for ESLint](https://github.com/testing-library/eslint-plugin-testing-library), extending their Recommended and React configurations.

This is so that projects that use this library (such as SumUp's Partner Portal) can get better linter feedback.

Please advise, I'm not sure if I added everything correctly!

## Definition of done

- [x] Development completed
- [x] Reviewers assigned
- [x] Unit and integration tests

### CODE REVIEW

**Comment by larimaza** [comment]
Should I sign the contributor agreement or just be added as a member?

**Comment by connor-baer** [comment]
@larimaza You're already a member :) The bot isn't smart enough yet to know who's a SumUp employee. [Contributions to teaching it](https://github.com/sumup/oss-clark) are welcome 😄

No need to sign unless you want to.

**Comment by connor-baer** [comment]
@larimaza Regarding the failing tests: can you make sure you're using the required version of Node ([v10.x](https://github.com/sumup-oss/foundry/blob/main/.nvmrc)) and run the tests again, please?

File: src/configs/eslint/config.ts [hunk 1]
```diff
@@ -281,6 +281,14 @@ function customizeFramework(frameworks?: Framework[]): EslintConfig {
...
       ],
     },
+    [Framework.TESTING_LIBRARY]: {
+      extends: [
+        'plugin:testing-library/react',
+        'plugin:testing-library/recommended',
+      ],
+      plugins: ['testing-library'],
+      overrides: [{ files: ['**/*spec.*'] }],
+    },
```
Comment (connor-baer) [inline src/configs/eslint/config.ts:291]: The `extends` and `plugins` options only should/need to apply to spec files, could you move them into the `overrides`, please?

Looking at the existing configs, I realized that they were suboptimal as well, so I've opened a PR (https://github.com/sumup-oss/foundry/pull/245) to fix them, too.

**Review by connor-baer — CHANGES_REQUESTED** [review]
Thank you so much for this fantastic contribution! 🙌🏻

File: src/configs/eslint/config.ts (hunk 1 above)
Comment (larimaza) [reply src/configs/eslint/config.ts:291]: Ooh, THAT'S what `overrides` is for! 😅

**Review by connor-baer — APPROVED** [review]
✨

**Pull request merged**